cross-chain/arbitrum/deploy_l2/16_update_with_solana_in_wormhole_gateway_mapping.ts

```diff
@@ -0,0 +1,39 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts } = hre
+  const { execute, log } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  // Fake SolanaWormholeGateway for local development purposes only.
+  const fakeSolanaWormholeGateway =
+    "0x11a22dc2e01ecd2ae40864822d4406ff8aed4e2b8932385dabe818422ff67e1b"
+
+  // See https://docs.wormhole.com/wormhole/blockchain-environments/solana
+  // This ID is valid for both Solana Devnet and Mainnet
+  const solanaWormholeChainID = 1
+
+  const solanaWormholeGateway = await deployments.getOrNull(
+    "SolanaWormholeGateway"
+  )
+
+  let solanaWormholeGatewayAddress = solanaWormholeGateway?.address
+  if (!solanaWormholeGatewayAddress && hre.network.name === "hardhat") {
+    solanaWormholeGatewayAddress = fakeSolanaWormholeGateway
+    log(`fake SolanaWormholeGateway address ${solanaWormholeGatewayAddress}`)
+  }
+
+  await execute(
+    "ArbitrumWormholeGateway",
+    { from: deployer, log: true, waitConfirmations: 1 },
+    "updateGatewayAddress",
+    solanaWormholeChainID,
+    solanaWormholeGatewayAddress
+  )
+}
+
+export default func
+
+func.tags = ["SetSolanaGatewayAddress"]
+func.dependencies = ["ArbitrumWormholeGateway"]
```

Comment:
Can we define both contracts to be resolved as dependencies? Can we update it similarly in other scripts?

```suggestion
func.dependencies = ["ArbitrumWormholeGateway", "SolanaWormholeGateway"]
```